### ISSUE TITLE
small things

### ISSUE THREAD

**Issue by kalmarek** [kalmarek/SymbolicWedderburn.jl#5]
I have a few syntactic comments on `eigenspacedecomposition.jl`. The real work is done there in `left_eigen` and `row_echelon_form!`. These need to be strictly typed, but
* we should use `AbstractMatrix` rather than `Matrix`.
* `row_echelon_form`, `[right|left]_nullspace`, `eigen_decomposition!`, etc. should take `AbstractMatrix` (without `T`)
* `normalize` is never used (and is a very strange function, semantically :)
* (in `_find_l`) I've never seen `!(j isa Nothing)`. Is there a benefit over `j !== nothing` (which I believe is more standard? correct me if I'm wrong)
* both of the occurrences of `findfirst` can be called on `@view`.

But really thanks for writing those!

**Comment by tweisser** [comment]
Sounds good. I don't understand the `@view` part though....

**Comment by kalmarek** [comment]
e.g. ```julia
j = findfirst(x -> !iszero(x), A[pos+1:end, i])
```
this allocates unnecessary vector `A[pos+1:end, i]`. `@view A[pos+1:end, i]` will not allocate, at least if `findfirst` is inlined (and will not allocate unconditionally on julia 1.5)

**Comment by tweisser** [comment]
Ah! I see. I added this. Are the signatures in esd alright, now?

**Comment by kalmarek** [comment]
did you push your changes? ;)

**Comment by tweisser** [comment]
Did you check out the smallthings branch? ;)

**Comment by kalmarek** [comment]
no I didn't;) I was waiting for PR :)

**Comment by kalmarek** [comment]
closed by #6
